opentelemetry/proto/agent/common/v1/common.proto

```diff
@@ -95,3 +95,20 @@ message ServiceInfo {
 
   // TODO(songya): add more fields as needed.
 }
+
+// ExportResultCode is the result code returned in the export responses.
+enum ExportResultCode {
```

Comment:
I remember you told me that this is controlled by the status we return from grpc not by the message.

Reply:
grpc status is used for throttling purposes: https://github.com/open-telemetry/oteps/blob/master/text/0035-opentelemetry-protocol.md#throttling

ResultCode is different and is an explicit field in the response.

The reason I am using grpc code for throttling is that there is already clearly defined grpc code for this (Unavailable). As opposed to that there is no grpc code that allows to distinguish between FailedRetryable and FailedNotRetryable cases and I don't want to artificially model them using other grpc codes.

Reply:
But from the Status you can determine if a retry is possible or not correct?

Reply:
Is there a clearly defined condition / set of status codes that allow client to make a decision about retrying or not retrying based just the grpc status code? If there is such a way then we can eliminate ResultCode.

At the moment I am not sure how a client should make such decision by looking at the returned gRPC code. For example if gRPC Status is `Unknown` is it retryable or no?

Perhaps instead of ResultCode we list here the status codes which must be interpreted by the client as Retryable and then make a recommendation for server to return specific errors codes to Retryable and NonRetryable cases? That could work.

Reply:
FYI: So you always get a status, but I think you never get a response + an error status (if I remember correctly).

So the logic I would use retry if:
* If the status contains the RetryInfo is a retryable 
* For some status codes without a RetryInfo (deadline exceeded, need to determine the full list) you may do retry.

Reply:
>FYI: So you always get a status, but I think you never get a response + an error status (if I remember correctly).

Yes, that is correct. A response only comes when status is success. Any other status is received without a response.

>If the status contains the RetryInfo is a retryable

So you are saying RetryInfo with RetryDelay=0 would be equivalent to ResultCode=FailedRetryable. The same RetryInfo with RetryDelay>0 would be throttling indication (also retryable obviously, but after a delay).

>For some status codes without a RetryInfo (deadline exceeded, need to determine the full list) you may do retry.

Make sense. Also, we need to decide which status code to return to indicate NotRetryable. Let me think though and submit an update.

Reply:
I created https://github.com/open-telemetry/oteps/pull/65
If it gets accepted I will discard this PR.